client/src/components/search/component.tsx

```diff
@@ -1,38 +1,43 @@
-import { FC, useRef } from 'react';
+import { FC, useCallback, useState } from 'react';
 import { SearchIcon, XIcon } from '@heroicons/react/solid';
-import { useButton } from '@react-aria/button';
-import { useSearchField } from '@react-aria/searchfield';
-import { useSearchFieldState } from '@react-stately/searchfield';
 
 import Input from 'components/forms/input';
 
 import type { SearchProps } from './types';
 
-export const Search: FC<SearchProps> = ({ ...props }: SearchProps) => {
-  const { placeholder } = props;
-  const state = useSearchFieldState(props);
+export const Search: FC<SearchProps> = (props: SearchProps) => {
+  const { onChange } = props;
+  const [value, setValue] = useState<SearchProps['value']>(null);
```

Comment:
Curious, why keep the value in the internal state instead of this being a controlled component? It's supposed to accept a `value` prop 🤔

Reply:
It's because this component is not only a single input, I wanted a embedded behavior of the reset button